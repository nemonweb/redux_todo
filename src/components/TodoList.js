import React, { Component } from 'react'
import TodoItem from './TodoItem'

export default class TodoList extends Component {
  render() {
    return (
      <ul>
        {this.props.todos.map((item, index) => (
          <TodoItem
            key={index}
            index={index}
            text={item.text}
            completed={item.completed}
            onClick={this.props.onTodoClick}
          />
        ))}
      </ul>
    )
  }
}
