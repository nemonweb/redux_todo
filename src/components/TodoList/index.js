import React, { Component } from 'react'
import TodoItem from '../TodoItem'
import PropTypes from 'prop-types'

export default class TodoList extends Component {
  render() {
    return (
      <ul>
        {this.props.todos.map((item, index) => (
          <TodoItem
            key={index}
            index={index}
            text={item.text}
            completed={item.completed}
            onClick={this.props.onTodoClick}
          />
        ))}
      </ul>
    )
  }
}

TodoList.propTypes = {
  todos: PropTypes.array.isRequired,
  onTodoClick: PropTypes.func.isRequired,
}
