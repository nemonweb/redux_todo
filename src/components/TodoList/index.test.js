import React from 'react'
import TodoList from './index'

describe('TodoList', () => {
  const todos = [{ completed: false, text: 'text' }]
  const onClickHandler = jest.fn()

  it('should properly render self when has todos', () => {
    const wrapper = shallow(<TodoList todos={todos} onTodoClick={onClickHandler} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should properly render self when todos empty', () => {
    const wrapper = shallow(<TodoList todos={[]} onTodoClick={onClickHandler} />)
    expect(wrapper).toMatchSnapshot()
  })
})
