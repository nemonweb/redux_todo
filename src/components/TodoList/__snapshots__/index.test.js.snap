// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList should properly render self when has todos 1`] = `
<ul>
  <TodoItem
    completed={false}
    index={0}
    key="0"
    onClick={[Function]}
    text="text"
  />
</ul>
`;

exports[`TodoList should properly render self when todos empty 1`] = `<ul />`;
